torch/csrc/distributed/c10d/ProcessGroupGloo.hpp

```diff
@@ -94,6 +94,7 @@ class TORCH_API ProcessGroupGloo : public ProcessGroup {
 
     const std::vector<std::vector<at::Tensor>> outputTensors_;
     c10::intrusive_ptr<at::ivalue::Future> future_;
+    std::function<void()> recordFunctionBeforeCallback_;
```

Comment:
Do we need to initialize this to some non-truthy value or is it done by default? Also would be good to add a comment explaining this field.

Reply:
Yep, it is initialized by default.